Task 2/Task 2.1.2/Task 2.1.2/Program.cs

```diff
@@ -0,0 +1,804 @@
+                    CustomPaint(name, figures);
+                    break;
+            }
+            Console.ReadKey();
+        }
+
+        /// <summary>
+        /// Метод для вывода основного меню.
+        /// </summary>
+        static void showMainMenu ()
+        {
+            List<string> menu = new List<string>()
+            {
+                "Выберите действие:",
+                "1. Добавить фигуру",
+                "2. Вывести фигуры",
+                "3. Очистить холст",
+                "4. Сменить пользователя",
+                "5. Выход"
+            };
+
+            foreach (string item in menu)
+            {
+                Console.WriteLine(item);
+            }
+        }
+
+        /// <summary>
+        /// Метод, выводящий меню выбора фиугр.
+        /// </summary>
+        static void ShowFiguresMenu ()
+        {
+            foreach (Figures item in Enum.GetValues(typeof(Figures)))
+            {
+                Console.WriteLine($"{(int)item}. {item}");
+            }
+        }
+
+        /// <summary>
+        /// Метод, использощуйся для проверки, введенного пользователем значения при выборе фигуры.
+        /// Если пользователь ввел значение не соответствующее интовому значению из перечисления Figures, 
+        /// метод будет просить переввести это значение.
+        /// </summary>
+        /// <returns>Корректное значение</returns>
+        public static int CheckUsersChoise ()
+        {
+            string uservalue = Console.ReadLine();
+            Int32.TryParse(uservalue, out int value);
+            while (value <= 0 || value > 8)
+            {
+                Console.WriteLine("Вы ввели некорректное значение. Попробуйте еще раз");
+                uservalue = Console.ReadLine();
+                Int32.TryParse(uservalue, out value);
+            }
+            return value;
+        }
+
+        /// <summary>
+        /// Метод, отвечающий за выбор фигуры.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns>Возвращает выбранную из перечисления фигуру. Тк значение которое попадает в этом метод уже прошло проверку на корректность,
+        /// то строчка "return Figures.Кольцо;" никогда не должна выполниться.
+        /// </returns>
+        public static Figures CheckFigure (int value)
+        {
+            foreach (Figures item in Enum.GetValues(typeof(Figures)))
+            {
+                if ((int)item == value)
+                {
+                    return item;
+                }
+            }
+
+            return Figures.Кольцо;
+        }
+
+        /// <summary>
+        /// Метод, который создает фигуры, в зависимости от выбора пользователя.
+        /// </summary>
+        static void ShowFigures (string name, List<object> figures, Figures figure)
+        {
+            switch (figure)
+            {
+                case Figures.Круг:
```

Comment:
Did you like to switch keyboard layouts? What would your american colleague do about this?